### DISCUSSION TITLE
After upgraded to the netbox latest versions to v3.5.1

### DISCUSSION THREAD

**Discussion by littealvin** [netbox-community/netbox#12576] [Help Wanted!]
Hello, It should be ok after I checked the dependencies to required versions as below. 
https://github.com/netbox-community/netbox/blob/develop/docs/installation/upgrading.md

However, I got the server error after upgraded as below. It would be much appreciated if anyone can help on this. 

Server Error
There was a problem with your request. Please contact an administrator.
The complete exception is provided below:

<class 'django.contrib.contenttypes.models.ContentType.DoesNotExist'>
ContentType matching query does not exist.

Python version: 3.8.10
NetBox version: 3.5.1

If further assistance is required, please post to the NetBox discussion forum on GitHub

**Comment by abhi1693** [comment]
Can you provide more stacktrace on the error as it is not possible to help with the provided message?

**Reply by Sawborg** [reply]
I got the same error when I upgraded tonight. My Netbox is borked. Haven't had issues in the past upgrading by cloning the git repo and running the upgrade script. This is what I get when I restart the services and open the URL to my login page. I believe I was running Netbox v3.2 or v3.3. 

![Screenshot 2023-05-15 at 9 26 08 PM](https://github.com/netbox-community/netbox/assets/17443788/1de6b47f-81be-492f-9be2-8832a2dabd88)

I ran the migration script manually but get errors when doing so.  I stopped the server, restored from a backup that I took right before the upgrade, restarted the VM instance and I'm still getting the same error.

**Comment by Sawborg** [comment]
So I managed to fix this. I restored from my backup and tried the upgrade from the git repo again. I was getting the following error during the upgrade: 

django.db.utils.DataError: Unable to proceed with deleting contact fields from Provider model: Found 8 providers with legacy contact data. Please ensure all legacy provider contact data has been migrated to contact objects before proceeding. Or, set the NETBOX_DELETE_LEGACY_DATA environment variable to bypass this safeguard and delete all legacy provider contact data.

I forced the legacy data deletion in the upgrade script: sudo NETBOX_DELETE_LEGACY_DATA=1 ./upgrade.sh

Restarted Netbox: sudo systemctl restart netbox netbox-rq

And we're back!
